DapperDoodle/QueryExecutor.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using DapperDoodle.Interfaces;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace DapperDoodle
+{
+    public class QueryExecutor : IQueryExecutor
+    {
+        public IBaseSqlExecutorOptions Options { get; }
+
+        public QueryExecutor(IServiceProvider services)
+        {
+            Options = services.GetService<IBaseSqlExecutorOptions>();
+        }
+        
+        public void Execute(Query query)
+        {
+            query.InitialiseDependencies(Options);
+            ExecuteWithNoResult(query);
+        }
+
+        private void ExecuteWithNoResult(Query query)
+        {
+            if (query == null)
+            {
+                throw new ArgumentNullException(nameof(query));
+            }
+
+            query.QueryExecutor = this;
+            query.Execute();
+        }
+
+        public T Execute<T>(Query<T> query)
+        {
+            query.InitialiseDependencies(Options);
+            ExecuteWithNoResult(query);
+            return query.Result;
+        }
+
+        public void Execute(IEnumerable<Query> queries)
+        {
+            if (queries == null)
```

Comment:
@RalphMcDougall I didn't realize the comment you posted was a reply on my comment, so when I deleted my comment it deleted yours as well. Sorry about that.

*This is what was posted before for reference*

If one has a list of queries, the following options are equivalent
                
``` csharp
                Execute(listOfQueries);
                for (q in listOfQueries)
                    results.append(q.Result)
                
                for (q in listOfQueries)
                    results.append(Execute(q))
                
                results = Execute(listOfQueries)
```

the second option is more concise than the first, so it removes the desire to be able to execute multiple queries.
                Allowing for the third case would be the most concise and probably ideal for this use-case?

Reply:
@RalphMcDougall Just to leave a reply on this comment. It is a good idea, but if I understand what you mean in the above comment correctly the reason it isn't feasible is because the result is part only accessed from QueryExecutor. It cannot be appended as a list. If we wanted to return a list we can create a method for that. But in this case `Execute(query)` is just a void method that doesn't return anything. So Result doesn't receive a value. This can be fixed, but to what end? What do you want to achieve by allowing the user to pass in several queries and expect a value in return. Maybe the feature should be dropped altogether, because since it is void a command would achieve the same thing as a query, so you make a good point in that regard.

Reply:
Yeah, I'm wondering how this method is suppose to be used. Unlike insert/delete, queries are useful purely because they return a value that one can use further. Since this method takes in a list of queries, I would have expected a list of results to be produced, but since nothing is returned it doesn't seem particularly useful for executing multiple queries. This is specifically for select statements, but this method might be used for inserts/deletes as well in which case it's probably fine.

Reply:
@RalphMcDougall Yeah you are right. I think I'm going to remove this feature. It doesn't really make sense when you think about it. If you want to want a list of values returned from your query then you would specify that in the generic parameter so that you receive the result. This feature this seems like bloated uselessness. I don't know what I was thinking.

Thanks for the feedback!